Sources/Build/BuildDescription/SwiftTargetBuildDescription.swift

```diff
@@ -535,6 +535,22 @@ public final class SwiftTargetBuildDescription {
             args += ["-color-diagnostics"]
         }
 
+        // If this is a generated test discovery target, it might import a test
+        // target that is built with C++ interop enabled. In that case, the test
+        // discovery target must enable C++ interop as well
+        switch testTargetRole {
+        case .discovery:
+            for dependency in try self.target.recursiveTargetDependencies() {
+                let dependencyScope = self.buildParameters.createScope(for: dependency)
+                let dependencySwiftFlags = dependencyScope.evaluate(.OTHER_SWIFT_FLAGS)
+                if let interopModeFlag = dependencySwiftFlags.first(where: { $0.hasPrefix("-cxx-interoperability-mode=") }) {
```

Comment:
are we doing these kind of checks elsewhere in the code? should it be encapsulated behind something like `dependencySwiftFlags.cxxInteropMode` returning an optional?

Reply:
Generalizing `extraSwiftFlags.first(where:)` makes sense to me, the other fine seems fine as is.

Reply:
Hmm, I just realized that the other usage of `extraSwiftFlags.first` operates on `[String]`, not on flags/scopes. So I don't really see a way to simplify this.